enterprise/mqtt/src/main/java/io/crate/mqtt/protocol/MqttProcessor.java

```diff
@@ -159,12 +151,8 @@ public void accept(Object result, Throwable t) {
         }
 
         private static void sendPubAck(Channel channel, int packetId, boolean isDupFlag) {
-            channel.writeAndFlush(MqttMessageBuilders.pubAck()
-                    .qos(MqttQoS.AT_LEAST_ONCE)
-                    .isDup(isDupFlag)
-                    .packetId(packetId)
-                    .build())
-                    .addListener(cf -> LOGGER.trace("PUBACK sent"));
+            channel.writeAndFlush(MqttMessageFactory.newPubAckMessage(MqttQoS.AT_LEAST_ONCE, isDupFlag, packetId))
+                .addListener(cf -> LOGGER.trace("PUBACK sent"));
```

Comment:
I'm not sure if it's worth it performance wise, but in `Messages` we adopted the pattern to just add a listener if the logging is required:

```
        ChannelFuture channelFuture = channel.writeAndFlush(buffer);
        if (LOGGER.isTraceEnabled()) {
            channelFuture.addListener((ChannelFutureListener) future -> LOGGER.trace("sentReadyForQuery"));
        }
```

Btw. is it from a protocol perspective necessary to flush on all these messages immediately? That has a quite big performance impact.